server/sonar-server/src/main/java/org/sonar/server/es/NewIndex.java

```diff
@@ -175,6 +176,7 @@ public void build() {
     private final NewIndexType indexType;
     private final String fieldName;
     private boolean sortable = false, wordSearch = false, gramSearch = false, docValues = false, disableSearch = false;
+    private SortedMap<String, Object> subFields = Maps.newTreeMap();
```

Comment:
I got the definitive explanation by @simonbrandhof : we want to be able to detect changes in the settings of an index, to drop and rebuild it when necessary. The detection of modifications is based on an order sensitive hashing algorithm (SHA256), so order of settings must be preserved - hence the use of SortedMap. I will enforce it in the ``addSubField`` method.